xtable-core/src/main/java/org/apache/xtable/parquet/ParquetStatsExtractor.java

```
@@ -0,0 +1,139 @@
+            List<ColumnChunkMetaData> columns = blockMetaData.getColumns();
+            columnDescStats =
+                    columns
+                            .stream()
+                            .collect(Collectors.groupingBy(columnMetaData -> schema.getColumnDescription(columnMetaData.getPath().toArray()),
+                                    Collectors.mapping(columnMetaData ->ColumnStat.builder()
+                                            .numValues(columnMetaData.getValueCount())
+                                            .totalSize(columnMetaData.getTotalSize())
+                                            .range(Range.vector(columnMetaData.getStatistics().genericGetMin(), columnMetaData.getStatistics().genericGetMax()))
+                                            .build(), Collectors.toList())));
+        }
+        return columnDescStats;
+    }
+
+    private InputPartitionFields initPartitionInfo() {
+        return partitions;
+    }
+
+    private InternalDataFile toInternalDataFile(
+            Configuration hadoopConf, Path parentPath) throws java.io.IOException {
+        FileStatus file = null;
+        List<PartitionValue> partitionValues = null;
+        ParquetMetadata footer = null;
+        List<ColumnStat> columnStatsForAFile = null;
+        try {
+            FileSystem fs = FileSystem.get(hadoopConf);
+            file = fs.getFileStatus(parentPath);
+            InputPartitionFields partitionInfo = initPartitionInfo();
+            footer = parquetMetadataExtractor.readParquetMetadata(hadoopConf, parentPath);
+            MessageType schema = parquetMetadataExtractor.getSchema(footer);
+            columnStatsForAFile = getColumnStatsForaFile(footer);
+            partitionValues = partitionExtractor.createPartitionValues(
+                    partitionExtractor.extractPartitionValues(
+                            partitionInfo));
+        } catch (java.io.IOException e) {
+
```

This should throw an exception since it would be an unexpected state.

---

what is exception type? is there already a defined custom one?

---

We can use `org.apache.xtable.exception.ReadException`